src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,241 @@
+   */
+  warnText?: React.ReactNode;
+
+}
+
+// Weekdays shorthand for english locale for flatpickr
+l10n.en.weekdays.shorthand.forEach((_day, index) => {
+  const currentDay = l10n.en.weekdays.shorthand;
+  if (currentDay[index] === 'Thu' || currentDay[index] === 'Th') {
+    currentDay[index] = 'Th';
+  } else {
+    currentDay[index] = currentDay[index].charAt(0);
+  }
+});
+
+const DatePicker = React.forwardRef((
+  {
+    allowInput = false,
+    className,
+    defaultValue = [],
+    disabled,
+    enableTime = false,
+    hideLabel,
+    id,
+    inline,
+    invalid,
+    invalidText,
+    labelText,
+    locale = 'en',
+    onChange = noOP,
+    onClick = noOP,
+    placeholder,
+    readOnly,
+    size = 'md',
+    type = 'range',
+    value,
+    warn,
+    warnText,
+    ...rest
+  }: DatePickerProps,
+  ref:React.ForwardedRef<HTMLInputElement>
+) => {
+  const inputProps = useNormalizedInputProps({disabled, id, invalid, invalidText, readOnly, type, warn, warnText})
+  const fp = React.useRef() as React.MutableRefObject<flatpickr.Instance>;
+  const inputRef = React.useRef() as React.Ref<HTMLInputElement> | undefined;
+  const manualValue = React.useRef<Date[]>()
+  React.useEffect(
+    () => {
+      // Config for flatpickr
+      const config: flatpickr.Options.Options = {
+        allowInput,
+        clickOpens: !inputProps.disabled,
+        defaultDate: defaultValue,
+        enableTime,
+        locale: l10n[locale as keyof typeof l10n],
+        mode: type,
+        onChange,
+      }
+      // if Ref is provided use
+      if(typeof ref === 'object' && ref?.current) {
+        fp.current = flatpickr(ref.current, config);
+      }
+      // No ref provided by users so use local ref
+      if(ref === null) {
+        if(typeof inputRef === 'object' && inputRef?.current) {
+          fp.current = flatpickr(inputRef?.current, config);
+        }
+      }
+      return () => {
+        if (fp.current && fp.current.destroy) {
+          fp.current.destroy();
+        }
+      }
+    },[allowInput, defaultValue, inputProps.disabled, enableTime, id, locale, onChange, ref, type]
+  )
+
+
+  const handleKeyPress = (e: KeyboardEvent) => {
+    if (allowInput && (/[0-9-/:]+/g).test(e.key) ) {
+      manualValue.current = fp?.current?.selectedDates
+    }
+  }
+  const handOnBlur = (e: FocusEvent) => {
+    if (
+      !allowInput ||
+      e.relatedTarget instanceof HTMLElement
+      && e.relatedTarget?.classList.value.includes('flatpickr')
+    ) {
+      return;
+    }
+    if (
+      'current' in manualValue
+      && manualValue?.current?.length
+      && manualValue?.current?.length > 0
+      && manualValue?.current?.join().toString() !== fp?.current?.selectedDates.join().toString()
+      ) {
+      const element = fp.current?.element as HTMLInputElement
+      manualValue.current = []
+      onChange(fp.current?.selectedDates, element?.value, fp.current)
+    }
+  }
```

Comment:
This is to deal with flatpickr not firing an onchange event from manual input unless you hit enter or tab away from the input.